glusterd2/commands/devices/adddevice.go

```diff
@@ -0,0 +1,110 @@
+package devicecommands
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/pborman/uuid"
+	"net/http"
+
+	"github.com/coreos/etcd/clientv3"
+	device "github.com/gluster/glusterd2/glusterd2/device"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/store"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/pkg/api"
+	log "github.com/sirupsen/logrus"
+)
+
+func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URLi
+	ctx := r.Context()
+	req := new(device.AddDeviceReq)
+	if err := restutils.UnmarshalRequest(r, req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "", api.ErrCodeDefault)
+		return
+	}
+	deviceinfo := device.Info{
+		Names:  req.Names,
+		PeerID: req.PeerID,
+	}
+	if req.PeerID == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Invalid Peer ID", api.ErrCodeDefault)
+		return
+	}
+
+	_, err := store.Store.Get(context.TODO(), deviceinfo.PeerID.String())
```

Comment:
What is this code getting from the store ? There's nothing at the key there.

Under what etcd key and namespace are devices being stored.